d3/query-graphs.js

```diff
@@ -592,6 +594,47 @@ function drawQueryTree(target, treeData) {
         update(d);
     }
 
+    // Dash tween to make the highlighted edges animate from start node to end node
+    var tweenDash = function() {
+        var l = this.getTotalLength();
+        var i = d3.interpolateString("0," + l, l + "," + l);
+        return function(t) {
+            return i(t);
+        };
+    };
+
+    // Transition used to highlight edges on mouseover
+    var edgeTransitionIn = function(path) {
+        path.transition()
+            .duration(duration)
+            .attr("opacity", 1)
+            .attrTween("stroke-dasharray", tweenDash)
+            .attr("d", diagonal);
+    };
+
+    // Transition to unhighlight edges on mouseout
+    var edgeTransitionOut = function(path) {
+        path.transition()
+            .duration(duration)
+            .attr("opacity", 0)
+            .attr("d", diagonal);
+    };
+
+    // Handler builder for mouseover and mouseout
+    var mouseHandler = function(transition) {
+        return function(d) {
+            var crosslinks = svgGroup.selectAll("path.crosslink-highlighted");
+
+            // Filter the edges to those connected to the current node
+            crosslinks.filter(function(dd) {
+                var sourceMatches = d.x === dd.source.x && d.y === dd.source.y;
```

Comment:
why are we matching based on the coordinates?
Can't we just match based on object identity, i.e. `d == dd.source`?

Reply:
Indeed.